### DISCUSSION TITLE
react-hook-form `onChange` handler

### DISCUSSION THREAD

**Discussion by pauls-akmentins** [react-hook-form/react-hook-form#11784] [Q&A]
Hey, can someone point me to the place in `react-hook-form` repo to their implementation of the `onChange` handler, that comes from them `register` function? I can not find it really.

I am interested, because I have one edge case, where I need to write my own handler and I wanted to understand, what other than updating value, they are doing.

I saw in the docs, that it also triggers re-evaluation of values validity, but I am not sure that this info is up to date, so just want to the the source, if possible.

Thanks

**Comment by sujinleeme** [comment]
You can  check `register` function logic [here](https://github.com/react-hook-form/react-hook-form/blob/7cdad77ae13190e6b491d9b4b30d89709229b52b/src/logic/createFormControl.ts#L997).
